pkg/util/utils.go

```diff
@@ -133,6 +149,33 @@ type JobResult struct {
 	TestGridUrl                     string  `json:"TestGridUrl"`
 }
 
+type BugzillaResult struct {
+	Name                            string  `json:"name"`
+	Job                             string  `json:"job"`
+	Failures                        int     `json:"failures"`
+	KnownFailures                   int     `json:"knownFailures"`
+	Successes                       int     `json:"successes"`
+	PassPercentage                  float64 `json:"PassPercentage"`
+	PassPercentageWithKnownFailures float64 `json:"PassPercentageWithKnownFailures"`
+	TestGridUrl                     string  `json:"TestGridUrl"`
+}
+
+type SortedBugzillaComponentResult struct {
+	Name string `json:"name"`
+
+	JobsFailed []BugzillaJobResult `json:"jobsFailed"`
+}
+
+type BugzillaJobResult struct {
+	JobName           string `json:"jobName"`
+	BugzillaComponent string `json:"bugzillaComponent"`
+
+	FailuresByThisBugzillaComponent int `json:"failuresByThisBugzillaComponent"`
+	// This one is phrased as a failure percentage because we don't know a success percentage since we don't know how many times it actually ran
+	// we only know how many times its tests failed.
```

Comment:
if we don't know how many times it ran, what is this a percentage of?

Reply:
> if we don't know how many times it ran, what is this a percentage of?

we can know failures percentage out of all runs, but we cannot know success percentages since we don't know how often the test runs

Reply:
My goal here is not to determine how often a particular test itself fails or how frequent a failure of a test is relative to how often it runs.  The goal is to measure impact to the average job run for a given job, so that's a percentage against all runs

Reply:
fair enough, but i think calling it a "failure percentage" is what lead to my confusion.  failureCausePercentage?